### PR TITLE
refactor: remove duplicate RPC namespaces from arguments

### LINKED ISSUE #3703: Throw better error when duplicate RPC namespaces are specified
### Describe the feature

Currently, if you try to run `reth --http --http.api trace,trace`, you will get:
```console
thread 'main' panicked at 'No conflicts: MethodAlreadyRegistered("trace_block")',
/Users/fineas/reth/crates/rpc/rpc-builder/src/lib.rs:855:35
```

which is a correct behaviour with a non-obvious error message. We need to throw a better self-explanatory error.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ungaro**
splits the input val by commas, trims each element, and collects them into a HashSet to automatically remove duplicates and joins them
 Should solve #3703

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
yep this should fix it since everything is case sensitive

File: bin/reth/src/args/rpc_server_args.rs [hunk 1]
```diff
@@ -477,14 +477,21 @@ impl TypedValueParser for RpcModuleSelectionValueParser {
...
+            .map(str::trim)
+            .collect::<std::collections::HashSet<_>>()
+            .into_iter()
+            .collect::<Vec<_>>()
+            .join(", ")
+            .parse::<RpcModuleSelection>()
+            .map_err(|err| {
+                let arg = arg.map(|a| a.to_string()).unwrap_or_else(|| "...".to_owned());
+                let possible_values = RethRpcModule::all_variants().to_vec().join(",");
+                let msg = format!(
```
Comment (onbjerg) [inline bin/reth/src/args/rpc_server_args.rs:490]: hmm, it seems like splitting by `,` and trimming is already done in the `FromStr` impl of `RpcModuleSelection` - maybe we should just dedupe in there so this part of the code could just be `RpcModuleSelection::from_str(val)`?

ref https://github.com/paradigmxyz/reth/blob/d890053084e56872ed6415be7fa64e20c432d8be/crates/rpc/rpc-builder/src/lib.rs#L614C18-L625

Comment (shekhirin) [reply bin/reth/src/args/rpc_server_args.rs:490]: @ungaro pinging in case you missed it

**Comment by mattsse** [comment]
changed the fromstr slightly

**Pull request merged**